pinot-core/src/main/java/org/apache/pinot/core/plan/maker/InstancePlanMakerImplV2.java

```diff
@@ -248,6 +255,48 @@ public PlanNode makeSegmentPlanNode(IndexSegment indexSegment, QueryContext quer
     }
   }
 
+  private QueryContext rewriteExpressionsWithHints(IndexSegment indexSegment, QueryContext queryContext) {
+    List<ExpressionContext> newSelectExpressions = queryContext.getSelectExpressions();
+    if (newSelectExpressions != null && !newSelectExpressions.isEmpty()) {
+      newSelectExpressions = newSelectExpressions.stream().map(
```

Comment:
We can do in-place replacement to save garbage. Also suggest not using stream api for performance